R/compare_designs.R

```diff
@@ -0,0 +1,216 @@
+#'  compare_design_estimates(design1, design2)
+#'  compare_design_estimands(design1, design2)
+#' 
+#' @name compare_functions
+
+#' @rdname compare_functions
+#' @export
+compare_designs <- function(design1, design2, format = "ansi8", mode = "sidebyside", pager = "off", context = -1L) {
+  
+  code_comparison <- compare_design_code(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  data_comparison <- compare_design_data(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  estimands_comparison <- compare_design_estimands(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  estimates_comparison <- compare_design_estimates(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  structure(
+    list(code_comparison = code_comparison, data_comparison = data_comparison, 
+         estimands_comparison = estimands_comparison, estimates_comparison = estimates_comparison),
+    class = "design_comparison"
+  )
+  
+}
+
+#' @rdname compare_functions
+#' @export
+print.design_comparison <- function(x, ...) {
+  cat("Research design comparison\n\n")
+  
+  print_console_header("Compare design code")
+  
+  print(x$code_comparison)
+  
+  print_console_header("Compare draw_data(design)")
+  
+  print(x$data_comparison)
+  
+  print_console_header("Compare draw_estimands(design)")
+  
+  print(x$estimands_comparison)
+  
+  print_console_header("Compare draw_estimates(design)")
+  
+  print(x$estimates_comparison)
+}
+
+
+#' @rdname compare_functions
+#' @importFrom diffobj diffChr
+#' @export
+compare_design_code <- function(design1, design2, format = "ansi8", mode = "sidebyside", pager = "off", context = -1L) {
+  
+  design1 <- get_design_code(design1)
+  design2 <- get_design_code(design2)
+   
+  structure(
+    diffChr(
+      design1,
+      design2,
+      format = format,
+      mode = mode,
+      pager = pager,
+      context = context
+    ),
+    class = "Diff",
+    package = "diffobj"
+  )
+  
+}
+
+#' @rdname compare_functions
+#' @importFrom diffobj diffChr
+#' @export
+compare_design_summaries <- function(design1, design2, format = "ansi256", mode = "sidebyside", pager = "off", context = -1L) {
+  
+  seed <- .Random.seed
+  set.seed(seed)
+  design1 <- capture.output(summary(design1))
+  set.seed(seed)
+  design2 <- capture.output(summary(design2))
+  
+  structure(
+    diffChr(
+      design1,
+      design2,
+      format = format,
+      mode = mode,
+      pager = pager,
+      context = context
+    ),
+    class = "Diff",
+    package = "diffobj"
+  )
+}
+
+#' @rdname compare_functions
+#' @importFrom diffobj diffObj
+#' @export
+compare_design_data <- function(design1, design2, format = "ansi256", mode = "sidebyside", pager = "off", context = -1L) {
+  
+  seed <- .Random.seed
+  set.seed(seed)
+  design1 <- draw_data(design1)
+  set.seed(seed)
+  design2 <- draw_data(design2)
```

Comment:
One cute thing would be if the ending .Random.seed was the same after two draws - if not, then you know the designs had different access patterns to the RNG.